libs/wingc/src/struct_schema.rs

```diff
@@ -0,0 +1,60 @@
+use crate::{
+	ast::{Reference, Scope},
+	jsify::{JSifier, JSifyContext},
+	type_check::{is_udt_struct_type, resolve_user_defined_type, Struct},
+	visit::{self, Visit},
+	visit_context::VisitContext,
+};
+
+pub struct StructSchemaVisitor<'a> {
```

Comment:
Maybe this can be a more general phase like "pre-emit-visitor" or something along those lines. Naming is also hard

Reply:
Sounds like a pre-optimization to me. 🤓 Keeping this focusing around a single purpose (struct schemas) feels like the right call - if we end up with multiple pre-jsify passes, then we can abstract things when we get there

Reply:
I think you are right. Thanks for the sanity check!